### PR TITLE
Add `add_ons` to `pyproject.toml` in spaceflight starters

### PR DESCRIPTION
**Pull request by merelcht**
## Motivation and Context
While working on https://github.com/kedro-org/kedro/issues/3220 I noticed the spaceflight starters didn't have the `add_ons` field in the `pyproject.toml` yet. 

I only added it to the spaceflights starters, because those are the only ones that will be used in the new project creation flow.

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Assigned myself to the PR
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by noklam — APPROVED** [review]
thank you! Good catch

**Review by SajidAlamQB — APPROVED** [review]

**Comment by merelcht** [comment]
@SajidAlamQB and @noklam I re-requested your reviews to make you aware I also had to update the default value from "none" to "['none']" to be parsed correctly. Does that look right to you? Or do we want to change this so "none" would also be accepted?

**Review by SajidAlamQB — APPROVED** [review]

**Comment by noklam** [comment]
I think `none` should work as I expected this match the user prompt. How should I test this change?

File: spaceflights-pandas-viz/cookiecutter.json [hunk 1]
```diff
@@ -3,5 +3,5 @@
     "repo_name": "{{ cookiecutter.project_name.strip().replace(' ', '-').replace('_', '-').lower() }}",
     "python_package": "{{ cookiecutter.project_name.strip().replace(' ', '_').replace('-', '_').lower() }}",
     "kedro_version": "{{ cookiecutter.kedro_version }}",
-    "add_ons" : "none"
+    "add_ons": "['none']"
```
Comment (merelcht) [inline spaceflights-pandas-viz/cookiecutter.json:6]:
```suggestion
    "add_ons": "[]"
```

File: spaceflights-pandas/cookiecutter.json [hunk 2]
```diff
@@ -2,5 +2,6 @@
     "project_name": "Spaceflights Pandas",
     "repo_name": "{{ cookiecutter.project_name.strip().replace(' ', '-').replace('_', '-').lower() }}",
     "python_package": "{{ cookiecutter.project_name.strip().replace(' ', '_').replace('-', '_').lower() }}",
-    "kedro_version": "{{ cookiecutter.kedro_version }}"
+    "kedro_version": "{{ cookiecutter.kedro_version }}",
+    "add_ons": "['none']"
```
Comment (merelcht) [inline spaceflights-pandas/cookiecutter.json:6]:
```suggestion
    "add_ons": "[]"
```

File: spaceflights-pyspark-viz/cookiecutter.json [hunk 3]
```diff
@@ -3,5 +3,5 @@
     "repo_name": "{{ cookiecutter.project_name.strip().replace(' ', '-').replace('_', '-').lower() }}",
     "python_package": "{{ cookiecutter.project_name.strip().replace(' ', '_').replace('-', '_').lower() }}",
     "kedro_version": "{{ cookiecutter.kedro_version }}",
-    "add_ons" : "none"
+    "add_ons": "['none']"
```
Comment (merelcht) [inline spaceflights-pyspark-viz/cookiecutter.json:6]:
```suggestion
    "add_ons": "[]"
```

File: spaceflights-pyspark/cookiecutter.json [hunk 4]
```diff
@@ -3,5 +3,5 @@
     "repo_name": "{{ cookiecutter.project_name.strip().replace(' ', '-').replace('_', '-').lower() }}",
     "python_package": "{{ cookiecutter.project_name.strip().replace(' ', '_').replace('-', '_').lower() }}",
     "kedro_version": "{{ cookiecutter.kedro_version }}",
-    "add_ons": "none"
+    "add_ons": "['none']"
```
Comment (merelcht) [inline spaceflights-pyspark/cookiecutter.json:6]:
```suggestion
    "add_ons": "[]"
```

**Comment by merelcht** [comment]
> I think `none` should work as I expected this match the user prompt. How should I test this change? Note that this also doesn't match with https://github.com/kedro-org/kedro/blob/fd2151336afb5bf6ed792b336a07a7107e2e6703/kedro/framework/cli/starters.py#L219-L225C38 which return an `[]`. We need to be clear hear should it return `"none", ["none"] or []`. I think either `[]` or `["none"]` is fine.

Actually `none` doesn't match the user prompt. Any user prompts via `kedro new` or `kedro new --addons=none` results in `add_ons = []` in the `pyproject.toml`. However, "none" in `cookiecutter.json` will result in `add_ons = none`, which is invalid syntax. I've changed the default in `cookiecutter.json` to `"[]"` instead of `"none"`, because that results in `add_ons = []` which is the same as the CLI flow.

**Review by noklam — APPROVED** [review]

File: spaceflights-pandas-viz/{{ cookiecutter.repo_name }}/pyproject.toml [hunk 5]
```diff
@@ -37,6 +37,7 @@ namespaces = false
 package_name = "{{ cookiecutter.python_package }}"
 project_name = "{{ cookiecutter.project_name }}"
 kedro_init_version = "{{ cookiecutter.kedro_version }}"
+add_ons = {{ cookiecutter.add_ons | string | replace("\'", "\"") }}
```
Comment (SajidAlamQB) [inline spaceflights-pandas-viz/{{ cookiecutter.repo_name }}/pyproject.toml:40]: In core template I've changed add_ons in `pyproject.toml`:
```suggestion
add_ons = "{{ cookiecutter.add_ons | default('') | string | replace('\"', '\\\"') }}"
```

Not sure if this was correct approach?

Comment (SajidAlamQB) [reply spaceflights-pandas-viz/{{ cookiecutter.repo_name }}/pyproject.toml:40]: The ensures the default value is an empty string if no value is provided.

File: spaceflights-pandas/{{ cookiecutter.repo_name }}/pyproject.toml [hunk 6]
```diff
@@ -37,6 +37,7 @@ namespaces = false
 package_name = "{{ cookiecutter.python_package }}"
 project_name = "{{ cookiecutter.project_name }}"
 kedro_init_version = "{{ cookiecutter.kedro_version }}"
+add_ons = {{ cookiecutter.add_ons | string | replace("\'", "\"") }}
```
Comment (merelcht) [inline spaceflights-pandas/{{ cookiecutter.repo_name }}/pyproject.toml:40]:
```suggestion
add_ons = "{{ cookiecutter.add_ons | default('') | string | replace('\"', '\\\"') }}"
```

File: spaceflights-pyspark-viz/{{ cookiecutter.repo_name }}/pyproject.toml [hunk 7]
```diff
@@ -37,6 +37,7 @@ namespaces = false
 package_name = "{{ cookiecutter.python_package }}"
 project_name = "{{ cookiecutter.project_name }}"
 kedro_init_version = "{{ cookiecutter.kedro_version }}"
+add_ons = {{ cookiecutter.add_ons | string | replace("\'", "\"") }}
```
Comment (merelcht) [inline spaceflights-pyspark-viz/{{ cookiecutter.repo_name }}/pyproject.toml:40]:
```suggestion
add_ons = "{{ cookiecutter.add_ons | default('') | string | replace('\"', '\\\"') }}"
```

File: spaceflights-pyspark/{{ cookiecutter.repo_name }}/pyproject.toml [hunk 8]
```diff
@@ -37,6 +37,7 @@ namespaces = false
 package_name = "{{ cookiecutter.python_package }}"
 project_name = "{{ cookiecutter.project_name }}"
 kedro_init_version = "{{ cookiecutter.kedro_version }}"
+add_ons = {{ cookiecutter.add_ons | string | replace("\'", "\"") }}
```
Comment (merelcht) [inline spaceflights-pyspark/{{ cookiecutter.repo_name }}/pyproject.toml:40]:
```suggestion
add_ons = "{{ cookiecutter.add_ons | default('') | string | replace('\"', '\\\"') }}"
```

**Comment by merelcht** [comment]
> With this change this is the difference in CLI flow:
> 
> Before:
> 
> image
> 
> After:
> 
> image

Ooohh good catch!! We definitely don't want that. I'll revert the default change, and just keep the change to `pyproject.toml` to reflect what you did on the viz add on PR.

File: spaceflights-pandas-viz/cookiecutter.json [hunk 9]
```diff
@@ -3,5 +3,5 @@
     "repo_name": "{{ cookiecutter.project_name.strip().replace(' ', '-').replace('_', '-').lower() }}",
     "python_package": "{{ cookiecutter.project_name.strip().replace(' ', '_').replace('-', '_').lower() }}",
     "kedro_version": "{{ cookiecutter.kedro_version }}",
-    "add_ons" : "none"
+    "add_ons": "[]"
```
Comment (merelcht) [inline spaceflights-pandas-viz/cookiecutter.json:6]:
```suggestion
    "add_ons": "none"
```

File: spaceflights-pyspark-viz/cookiecutter.json [hunk 10]
```diff
@@ -3,5 +3,5 @@
     "repo_name": "{{ cookiecutter.project_name.strip().replace(' ', '-').replace('_', '-').lower() }}",
     "python_package": "{{ cookiecutter.project_name.strip().replace(' ', '_').replace('-', '_').lower() }}",
     "kedro_version": "{{ cookiecutter.kedro_version }}",
-    "add_ons" : "none"
+    "add_ons": "[]"
```
Comment (merelcht) [inline spaceflights-pyspark-viz/cookiecutter.json:6]:
```suggestion
    "add_ons": "none"
```

File: spaceflights-pyspark/cookiecutter.json [hunk 11]
```diff
@@ -3,5 +3,5 @@
     "repo_name": "{{ cookiecutter.project_name.strip().replace(' ', '-').replace('_', '-').lower() }}",
     "python_package": "{{ cookiecutter.project_name.strip().replace(' ', '_').replace('-', '_').lower() }}",
     "kedro_version": "{{ cookiecutter.kedro_version }}",
-    "add_ons": "none"
+    "add_ons": "[]"
```
Comment (merelcht) [inline spaceflights-pyspark/cookiecutter.json:6]:
```suggestion
    "add_ons": "none"
```

**Pull request merged**